### PR TITLE
#149: create endpoint for exporting users of current institution

### PR DESCRIPTION
**Pull request by DanBondarenko**
Resolves issue: https://github.com/keeleinstituut/tv-tolkevarav/issues/149

### CODE REVIEW

**Review by kadmit — APPROVED** [review]
@DanBondarenko Good job! Approved 👍

File: application/app/Policies/InstitutionUserPolicy.php [hunk 1]
```diff
@@ -99,6 +104,6 @@ public function apply(Builder $builder, Model $model): void
             abort(401);
         }
 
-        $builder->where('institution_id', $currentUserInstitutionId);
+        $builder->where('institution_id', $currentUserInstitutionId)->whereHas('user');
```
Comment (thenouan) [inline application/app/Policies/InstitutionUserPolicy.php:107]: Could you explain with a couple words why was this needed?

Comment (DanBondarenko) [reply application/app/Policies/InstitutionUserPolicy.php:107]: This effectively checks that the related User object has not been soft-deleted. Without this, the scope would include `institutution_user` records whose `user` has been soft-deleted.

Comment (thenouan) [reply application/app/Policies/InstitutionUserPolicy.php:107]: Okay, I guess that's fine for now. I was just thinking that it adds extra query every time the policy is applied, which can affect the query times. And the scope should answer the question "what objects of this resource can user see?" and this whereHas clause expands the answer to this question to "user can only see objects that have the related user object attached". I'm not sure if this is a authorization concern since I can think about cases such as 

> order has been created for a product, but since the product is not sold in the store anymore, then it's deleted, but it is still important for the business to see all orders that have been made with the sold products under with the order

In that sense it is more of a business logic decision and not a authorization concern. And I guess the same story is with users and institution users, there's also no business logic decision that the user object should ever be deleted and it wouldn't be wise to do as well, since user can be associated to multiple insitutions and the institution should only be able to interact with the institution user objects that are associated to this institution to mitigate issues that some user loses access to insitution1 when insitution2 deletes the user

**Review by thenouan — APPROVED** [review]

File: application/app/Policies/InstitutionUserPolicy.php (hunk 1 above)
Comment (DanBondarenko) [reply application/app/Policies/InstitutionUserPolicy.php:107]: > Okay, I guess that's fine for now. I was just thinking that it adds extra query every time the policy is applied, which can affect the query time
It doesn’t actually add any extra queries – it modifies the existing query, adding an exists clause (tested with Tinker + `DB::enableQueryLog();`).

> In that sense it is more of a business logic decision and not a authorization concern.
I see your point, especially with the example. Didn’t think of it that way before. I’ll remove it from the policy scope.
